sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/DataLoader.java

```diff
@@ -0,0 +1,122 @@
+        _client = Preconditions.checkNotNull(client,
+            "The CosmosAsyncClient needed for data loading can not be null");
+    }
+
+    public void loadData(final Map<Key, ObjectNode> records) {
+        bulkCreateItems(records);
+        validateDataCreation(records.size());
+    }
+
+    private void bulkCreateItems(final Map<Key, ObjectNode> records) {
+        final List<CosmosItemOperation> cosmosItemOperations = mapToCosmosItemOperation(records);
+        final String containerName = _configuration.getCollectionId();
+        final CosmosAsyncDatabase database = _client.getDatabase(_configuration.getDatabaseId());
+        final CosmosAsyncContainer container = database.getContainer(containerName);
+        LOGGER.info("Bulk loading {} documents in [{}:{}]", cosmosItemOperations.size(),
+            database.getId(),
+            containerName);
+
+        final BulkProcessingOptions<Object> bulkProcessingOptions = new BulkProcessingOptions<>(Object.class);
+        bulkProcessingOptions.setMaxMicroBatchSize(MAX_BATCH_SIZE)
+            .setMaxMicroBatchConcurrency(BULK_OPERATION_CONCURRENCY);
+        container.processBulkOperations(Flux.fromIterable(cosmosItemOperations), bulkProcessingOptions)
+            .blockLast(BULK_LOAD_WAIT_DURATION);
+
+        LOGGER.info("Completed document loading into [{}:{}]", database.getId(), containerName);
+    }
+
+    private void validateDataCreation(int expectedSize) {
+        final String containerName = _configuration.getCollectionId();
+        final CosmosAsyncDatabase database = _client.getDatabase(_configuration.getDatabaseId());
+        final CosmosAsyncContainer container = database.getContainer(containerName);
+        LOGGER.info("Validating {} documents were loaded into [{}:{}]",
+            expectedSize, _configuration.getDatabaseId(), containerName);
+
+        final List<FeedResponse<InternalObjectNode>> queryItemsResponseList = container
+            .queryItems(COUNT_ALL_QUERY, InternalObjectNode.class)
+            .byPage()
+            .collectList()
+            .block(BULK_LOAD_WAIT_DURATION);
+        final int resultCount = Optional.ofNullable(queryItemsResponseList)
+            .map(responseList -> responseList.get(0))
+            .map(FeedResponse::getResults)
+            .map(list -> list.get(0))
+            .map(internalObjectNode -> {
+                try {
+                    final ObjectNode result = internalObjectNode.getObject(ObjectNode.class);
+                    return result.get(COUNT_ALL_QUERY_RESULT_FIELD).intValue();
+                } catch (IOException ex) {
+                    LOGGER.error("Error extracting the result count from the response");
+                }
+                return 0;
+            })
+            .orElse(0);
+
+        if (resultCount != expectedSize) {
+            throw new IllegalStateException("Expected number of records " + expectedSize
```

Comment:
Good point. The reason I have this at exact match is because the following code assumes all the data in the Map is the DB (it uses the keys in the Map to randomly select the Keys).

Let's keep it for now, and I am open to changing it to say 95% tolerance if the record counts mismatch.